### PR TITLE
Update daily workflow to sharding

### PR DESCRIPTION
**Pull request by lanej0**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

Updates the daily testing workflows to execute end-to-end tests with sharding.

This PR also makes a fix to the reporting -- skipped tests have been reported higher than they should have because the Playwright blob reporter and allure reporters were conflicting.  This PR will correct that for all runs.

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Go to Actions > Daily smoke test
2. Run the workflow using this branch ([here](https://github.com/woocommerce/woocommerce/actions/runs/7807131110) is a sample run if you don't want to wait).
3. Ensure workflow passes -- k6 tests are failing on daily currently, so they fail here too.  We'll look at fixing that in a different PR.

### Changelog entry

-   [ ] Automatically create a changelog entry from the details below.

<details>

#### Significance

-   [ ] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Comment 

</details>

### CODE REVIEW

File: .github/workflows/smoke-test-daily.yml [hunk 1]
```diff
@@ -117,22 +128,67 @@ jobs:
...
-                  path: |
-                      ${{ env.ALLURE_RESULTS_DIR }}
-                      ${{ env.ALLURE_REPORT_DIR }}
-                  retention-days: 20
+                  name: all-blob-reports-${{ matrix.shard.number }}
+                  path: ${{ env.ALLURE_RESULTS_DIR }}
+                  retention-days: 1
+                  compression-level: 9
+            
+    e2e-tests-success:
```
Comment (adimoldovan) [inline .github/workflows/smoke-test-daily.yml]: I guess this job is not really needed, this workflow doesn't block anything being a scheduled one.

File: .github/workflows/smoke-test-daily.yml [hunk 2]
```diff
@@ -117,22 +128,67 @@ jobs:
...
+
+        - name: Download blob reports from GitHub Actions Artifacts
+          uses: actions/download-artifact@v4
+          with:
+              path: ${{ env.ALLURE_RESULTS_DIR }}
+              pattern: all-blob-reports-*
+              run-id: e2e-tests-run
+              merge-multiple: true
+
+        - name: Generate Playwright E2E Test report.
```
Comment (adimoldovan) [inline .github/workflows/smoke-test-daily.yml]: Nitpick, but `Generate Playwright E2E Test report.` makes me think the Playwright default report will be generated here, while in fact the Allure report is generated. If there was no Playwright report available it wouldn't be confusing. How about keeping it simple: `Generate test report`?

File: .github/workflows/smoke-test-daily.yml [hunk 3]
```diff
@@ -117,22 +128,67 @@ jobs:
...
+      runs-on: ubuntu-latest
+      needs: e2e-tests
+      if: success()
+      steps:
+        - run: echo Done!         
+    
+    merge-reports:
+      name: Merge e2e test reports
+      if: always()
+      needs: e2e-tests-success
```
Comment (adimoldovan) [inline .github/workflows/smoke-test-daily.yml]: The `e2e-tests-success` only runs on `success()`, so I guess this one will not run on tests failures.
But I think `e2e-tests-success` is not needed in this case (see my other comment) and this job should wait for `e2e-tests` instead.

**Review by adimoldovan — CHANGES_REQUESTED** [review]

File: .github/workflows/smoke-test-daily.yml (hunk 1 above)
Comment (lanej0) [reply .github/workflows/smoke-test-daily.yml]: Right. I was copying and pasting from the PR workflow :-)

**Comment by lanej0** [comment]
Thanks for your review @adimoldovan -- I've implemented all of your feedback, and fixed the last couple of bugs that were lingering with reporting.  The latest run can be seen here: https://github.com/woocommerce/woocommerce/actions/runs/7821375862

**Review by adimoldovan — APPROVED** [review]
Looks good 👍🏼

**Pull request merged**